gammapy/utils/fitting/parameter.py

```diff
@@ -528,6 +528,16 @@ def restore_values(self):
         """
         return restore_parameters_values(self)
 
+    @property
+    def frozen(self):
+        """Frozen parameters"""
+        return [par.frozen for par in self.parameters]
+
+    @frozen.setter
+    def frozen(self, value):
```

Comment:
The behavior is a bit strange no? This will freeze or thaw all parameters at once. If the property returns an array or a list, I would assume the setter to expect one as well. 

Wouldn't it be more clear to have a `freeze_all` or `thaw_all`methods?

Reply:
Thanks, that's a good point. I didn't realise the getter and setter approach was inconsistent. I'll implement you proposal with the `freeze_all()` method, which is the only one I need in this PR.